tests/graph/test_hydrate.py

```diff
@@ -52,6 +53,31 @@ class Tree(BaseModel):
     branches: list[Branch]
 
 
+class Caterpillar(BaseModel):
+    home: Leaf | None
+
```

Comment:
```suggestion


class Nature(BaseModel):
    flora_or_fauna: Tree | Caterpillar  # not supported
```

Reply:
added in 964bdfe22fe8646eeaa63f1e47b983c5e4f11d28